packages/stripe/src/lib/stripe.module.ts

```diff
@@ -39,34 +39,49 @@ const makeController = (prefix: string, route: string) => {
   return Controller;
 };
 
-export const stripeModule = new AppModule(
+export class StripeModule extends createModule(
   {
     config: stripeModuleConfig,
-    providers: [
-      {
-        provide: 'Stripe',
-        deps: [stripeModuleConfig.all()],
-        useFactory: (config: typeof stripeModuleConfig.type) => {
-          const { apiKey } = config;
-          return new Stripe(apiKey, {
-            typescript: true,
-            apiVersion: '2020-08-27',
-          });
-        },
-      },
-    ],
-    exports: ['Stripe'],
+    exports: [Stripe],
   },
   'stripe'
-).setup((stripeModule, config) => {
-  if (config.webhookConfig) {
-    const webhookConfig = config.webhookConfig!;
-    stripeModule.addProvider(StripeWebhookService);
-    stripeModule.addController(
-      makeController(
-        webhookConfig.controllerRoutePrefix!,
-        webhookConfig.controllerRoute!
-      )
-    );
+) {
+
+  process() {
+    this.addProvider({
+      provide: Stripe,
+      useFactory: () => {
+        return new Stripe(this.config.apiKey, {
+          typescript: true,
+          apiVersion: '2020-08-27',
+        });
+      },
+    });
+
+    if (this.config.webhookConfig) {
+      this.addProvider(StripeWebhookService);
+      this.addController(
+        makeController(
+          this.config.webhookConfig.controllerRoutePrefix,
+          this.config.webhookConfig.controllerRoute,
+          this.config.webhookConfig.secret
+        )
+      );
+    }
+  }
+
+  processController(module: AppModule<any>, controller: ClassType) {
+    if (!this.config.webhookConfig) return;
+
+    const configs = stripeHandlerMap.get(controller);
```

Comment:
I'm wondering about removing this `Map` in favor of attaching metadata to the parent class using the decorator which can then be checked here. This is the approach I used for the NestJS versions of some of my libraries and it felt nice that the information about what a Controller/Service could be used for was attached to it directly in the form of metadata